prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -87,6 +148,60 @@ class OktaAuthentication(private val minimumLevel: PrincipalLevel = PrincipalLev
         }
     }
 
+    /**
+     * Return a valid AuthenticatedClaims iff
+     * 1) The okta group in the claim is at least as high as the [requiredMinimumLevel]
+     * 2) The okta group in the claim matches the organizationName.
+     * Return null otherwise
+     */
+    fun authorizeByMembership(
```

Comment:
This used to be the ONLY Okta based authorization code.  

But different functions need to authorize based on different criteria, eg, in the case of this PR, the getReportHistory needs to authorize by making sure the report is "owned" by the organization in the user's claim.  (See method `checkSenderAccessAuthorization`, above)